packages/react-i18n/src/i18n.ts

```diff
@@ -632,64 +625,57 @@ export class I18n {
       day: 'numeric',
     });
   }
-}
 
-function normalizedNumber(
-  input: string,
-  decimalSymbol: string,
-  thousandSymbol: string,
-) {
-  const lastIndexOfNominalDecimal = input.lastIndexOf(decimalSymbol);
-  const lastIndexOfComma = input.lastIndexOf(COMMA);
-  const lastIndexOfPeriod = input.lastIndexOf(PERIOD);
-
-  let lastIndexOfRealDecimal = lastIndexOfNominalDecimal;
-
-  // For locales that do not use period as the decimal symbol, users may still input a period
-  // and expect it to be treated as the decimal symbol for their locale.
-  if (
-    decimalSymbol !== PERIOD &&
-    // where the thousand symbol is not a period
-    // and a period appears after the decimal symbol (or there is no decimal symbol)
-    ((thousandSymbol !== PERIOD &&
-      lastIndexOfPeriod > lastIndexOfNominalDecimal) ||
-      // where the thousand symbol is a period and a comma appears before a period
-      (thousandSymbol === PERIOD &&
-        lastIndexOfComma > -1 &&
-        lastIndexOfPeriod > -1 &&
-        lastIndexOfPeriod > lastIndexOfComma))
-  ) {
-    lastIndexOfRealDecimal = lastIndexOfPeriod;
-  }
+  private normalizedNumber(input: string, decimalSymbol: string) {
+    const lastIndexOfNominalDecimal = input.lastIndexOf(decimalSymbol);
+    const lastIndexOfPeriod = input.lastIndexOf(PERIOD);
 
-  const nonDigits = /\D/g;
-  const integerValue = input
-    .substring(0, lastIndexOfRealDecimal)
-    .replace(nonDigits, '');
-  const decimalValue = input
-    .substring(lastIndexOfRealDecimal + 1)
-    .replace(nonDigits, '');
+    let lastIndexOfRealDecimal = lastIndexOfNominalDecimal;
 
-  const isNegative = input.trim().startsWith('-');
-  const negativeSign = isNegative ? '-' : '';
+    // For locales that do not use period as the decimal symbol, users may still input a period
+    // and expect it to be treated as the decimal symbol for their locale.
+    if (
+      decimalSymbol !== PERIOD &&
+      (input.match(REGEX_PERIODS) || []).length === 1 &&
+      this.decimalValue(input, lastIndexOfPeriod).length <= 2
```

Comment:
I guess there are no locales that have groupings of less than 3 (for the least significant digits)?

(India uses groups of 2, but only after the first 3 least significant digits)

---

A hypothetical locale that uses period for grouping, digit groups of 2, and a decimal symbol of `,`...

In such a hypothetical, a number of `1234`, represented as `12.34,` would parse to `12.34`

I don't think such a grouping exists though (through my brief internet searching)

Reply:
I guess it also won't work for currencies that use more than 2 decimal digits (e.g., BHD uses three decimal digits; Bitcoin uses 8 decimal digits), or in scenarios where sub-cent accuracy is needed (e.g., banking typically uses 4 decimal digits)

e.g., A Baharaini-Frenchmen making a transfer of `123.50 BHD` would write `123.500`, but this would parse it as `123000 BHD`.

Reply:
> I guess there are no locales that have groupings of less than 3 (for the least significant digits)?

Not that I could see, but it's totally possible for me to have missed something.

Reply:
> e.g., A Baharaini-Frenchmen making a transfer of 123.50 BHD would write 123.500, but this would parse it as 123500 BHD?

Yes; in the French locale, how would we be able to guess if they mean 123500 or 123.500? In cases of ambiguity like this, I think it makes sense to fall back to locale formatting, e.g. `123,500`. 

Unfortunately, I think this would break the idea of being able to call `unformatCurrency` multiple times, because `123,500` would become `123.500`, which would become `123500.00`. I'm not sure how to get around this though, so open to ideas if you have any.

Reply:
I suppose, I could use the currency's number of decimal places to check for 3 or less instead of 2 or less if the currency's decimal places is 3. 

That would mean though, in the French locale and `BHD` currency scenario, that I would assume `123.500` to be `123.500`, not `123500`. If someone wanted to actually use `.` as a thousands separator when the number of places after the period is 3 or less, they would have to also supply the decimal places (e.g. `123.500,000`).

I don't know if this is any better though or if it just invites more potential problems.

Reply:
Note - I just created a second commit where I explore this scenario - please let me know what you think!

> I suppose I could use the currency's number of decimal places to check for 3 or less instead of 2 or less if the currency's decimal places is 3.
> 
> That would mean though, in the French locale and BHD currency scenario, that I would assume 123.500 to be 123.500, not 123500. If someone wanted to actually use . as a thousands separator when the currency's decimal places is 3, they would have to also supply the decimal separator (e.g. 123.500,000).